source/CMakeLists.txt

```diff
@@ -79,4 +79,34 @@ if(ADIOS_USE_MPI)
   target_include_directories(adios2 PUBLIC ${MPI_C_INCLUDE_PATH})
   target_compile_definitions(adios2 PUBLIC ADIOS_HAVE_MPI)
   target_link_libraries(adios2 PUBLIC ${MPI_C_LIBRARIES})
+else()
+  target_sources(adios2 PRIVATE mpidummy.cpp)
```

Comment:
This whole block looks like a bad merge.  The resulting CMakeLists.txt is borked.